src/centralstore/incremental/RelationBlock.h

```diff
@@ -0,0 +1,72 @@
+#include <cstring>
+#include <fstream>
+#include <set>
+#include <string>
+
+#include "NodeBlock.h"
+#include "PropertyLink.h"
+
+#ifndef RELATION_BLOCK
+#define RELATION_BLOCK
+struct NodeRelation {
+    unsigned int address = 0;
+    unsigned int nextRelationId = 0;
+    unsigned int nextPid = 0;
+    unsigned int preRelationId = 0;
+    unsigned int prePid = 0;
+};
+
+/**
+ * Relation states
+ *     Source       Destination
+ *       x               -
+ *       x               x
+ *       -               -
+ *       -               x
+ *
+ * **/
+
+class RelationBlock {
+   private:
+    std::string id;
+    bool updateRelationRecords(int recordType, unsigned int data);
+
+   public:
+    RelationBlock(unsigned int addr, NodeRelation source, NodeRelation destination, unsigned int propertyAddress)
+        : addr(addr), source(source), destination(destination), propertyAddress(propertyAddress){};
+
+    char usage;
+    unsigned int addr;  // Block size * block ID ID for this block
```

Comment:
why `ID ID` repeated?

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/115/commits/ed31ba0af33e9629173808960eef3cfe90c5dbe9

Reply:
Done